Lib/test/test_pdb.py

```diff
@@ -1066,6 +1077,65 @@ def test_post_mortem_from_none():
     """
 
 
+def test_post_mortem_from_no_stack():
+    """Test post mortem traceback debugging of chained exception
+
+    especially when one exception has not stack.
+
+    >>> def main():
+    ...     raise Exception() from Exception()
+
+
+    >>> def test_function():
+    ...     import pdb;
+    ...     instance = pdb.Pdb(nosigint=True, readrc=False)
+    ...     try:
+    ...         main()
+    ...     except Exception as e:
+    ...         # same as pdb.post_mortem(e), but with custom pdb instance.
+    ...         instance.reset()
+    ...         instance.interaction(None, e)
+
+    >>> with PdbTestInput(  # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
+    ...     ["exceptions",
+    ...     "exit"],
+    ... ):
+    ...    try:
+    ...        test_function()
+    ...    except ValueError:
+    ...        print('Correctly reraised.')
+    > <doctest test.test_pdb.test_post_mortem_from_no_stack[0]>(2)main()
+    -> raise Exception() from Exception()
+    (Pdb) exceptions
+    >   0 Exception()
+    (Pdb) exit
+    """
+
+
+def test_post_mortem_single_no_stack():
+    """Test post mortem called when origin exception has not stack
+
+
+    >>> def test_function():
+    ...     import pdb;
+    ...     instance = pdb.Pdb(nosigint=True, readrc=False)
+    ...     import sys
+    ...     sys.last_exc = Exception()
+    ...     # same as pdb.post_mortem(e), but with custom pdb instance.
+    ...     instance.reset()
+    ...     instance.interaction(None, sys.last_exc)
+
+    >>> with PdbTestInput(  # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
+    ...     []
+    ... ):
+    ...    try:
+    ...        test_function()
+    ...    except ValueError as e:
+    ...        print(e)
+    A valid traceback must be passed if no exception is being handled
```

Comment:
When a pdb user see this, are they necessarily in a context where this error message makes sense?

Isn't there some reasonable way to handle an empty traceback without erroring out? Is pm() just about inspecting the traceback or is there more to it?

If there needs to be an error message, I think it should just state what happened (rather than what the user should have done). So something like "No exception traceback to inspect".

Reply:
This is an already existing error of `post_mortem()`.

https://github.com/python/cpython/blob/40e52c94a27e4cd94b48e8a705914823cbb6afed/Lib/pdb.py#L2016-L2018

Except with the current logic in `post_mortem()` this does not trigger when we pass an exception to it and not a tb.

I'm happy to change it but wanted to stay as close as possible to the original.

So I think that yes, in this context someone getting the error message know what to expect. Up to you.

Reply:
This error is a bit wrong now then, if there is no exception being handled you need to pass a valid traceback or exception instance.

Anyway, the existing error is for another error case. I don't think we need to copy its wording.

Reply:
Changed to `"No exception traceback to inspect"`